### PR TITLE
Migrate from TSLint to ESLint

### PR DESCRIPTION
**Pull request by zFernand0**
:information_source: This PR is easier to review with whitespace changes hidden. It will still be a big diff though :cry:

In addition to the migration, this PR:
- Removes any tslint related rules and configuration files

Signed-off-by: zFernand0 <fernando.rijocedeno@broadcom.com>

### CODE REVIEW

**Comment by zFernand0** [comment]
Seems like there are a few system tests that fail on Windows due how FTP/MVS handles `//` compared to ZOSMF.

Failing system tests on windows:
- `__tests__/__system__/cli/view/uss-file/cli.view.uss-file.test.ts`
  - `should be able to upload a file to a uss directory then view it`
  - `should be able to upload to a uss file in binary mode, then view the content with the CLI and verify that the content is correct`
- ` __tests__/__system__/cli/delete/uss-file/cli.delete.uss-file.test.ts`
  - `should be able to mkdir a directory tree on uss then delete them recursively`

File: __tests__/__system__/cli/download/uss-file/cli.download.uss-file.test.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ describe("submit job from local file command", () => {
             [destination, downloadFilePath]);
         expect(result.stderr.toString()).toEqual("");
         expect(result.status).toEqual(0);
-        expect(IO.existsSync(downloadFilePath));
+        expect(IO.existsSync(downloadFilePath)).toBe(true);
```
Comment (t1m0thyj) [inline __tests__/__system__/cli/download/uss-file/cli.download.uss-file.test.ts:63]: It's cool that ESLint can find things like this to help us write better tests 👍

**Review by t1m0thyj — APPROVED** [review]

File: src/api/CreateDataset.ts [hunk 2]
```diff
@@ -185,12 +181,12 @@ export class CreateDataset extends AbstractTemplatedJCL {
         const jcl = this.getJcl(jobCardFile, {createDD}, overrideTemplateFile);
         this.log.debug("Creating data set %s via JCL:\n %s", dataSetName, jcl);
         const jobid = await
-            connection.submitJCL(jcl);
+        connection.submitJCL(jcl);
```
Comment (std4lqi) [inline src/api/CreateDataset.ts:184]: We may take this chance to put them into one line?

**Review by std4lqi — APPROVED** [review]
Just one minor comments. Thanks for this laborious work!!!

**Comment by tiantn** [comment]
Hi @zFernand0 , both unit tests and system tests passed. Thank you very much!

**Review by tiantn — APPROVED** [review]

**Pull request merged**